scripts/building/check_enforcer.ps1

```diff
@@ -0,0 +1,181 @@
+        "sdk_cli_azure_test" = 4;
+    }
+    $pipelines_count = @{
+        "sdk_cli_tests" = 0;
+        "sdk_cli_global_config_tests" = 0;
+        "sdk_cli_azure_test" = 0;
+    }
+
+    $(gh api /repos/$github_repository/commits/$MergeCommit/check-suites) `
+    | ConvertFrom-Json `
+    | Select-Object -ExpandProperty check_suites `
+    | ForEach-Object {
+        if ($snippet_debug -eq 1) {
+            Write-Output "check-suites id $($_.id)"
+        }
+        $suite_id = $_.id
+        $(gh api /repos/$github_repository/check-suites/$suite_id/check-runs) `
+        | ConvertFrom-Json `
+        | Select-Object -ExpandProperty check_runs `
+        | ForEach-Object {
+            if ($snippet_debug -eq 1) {
+                Write-Output "check runs name $($_.name)"
+            }
+            foreach ($key in $pipelines.Keys) {
+                $value = $pipelines[$key]
+                if ($value -eq 0) {
+                    continue
+                }
+                if ($_.name.Contains($key)) {
+                    $pipelines_count[$key] += 1
+                    $valid_status_array_ref.Value += $_
+                }
+            }
+        }
+    }
+    # Get pipelines of commit. count should match.
+    foreach ($key in $pipelines.Keys) {
+        if ($pipelines_count[$key] -lt $pipelines[$key]) {
+            $failed_reason_ref.Value = "Not all pipelines are triggered."
+        }
+    }
+}
+function sdk_cli_checks([ref]$failed_reason_ref, $valid_status_array) {
+    $failed_reason_ref.Value = ""
+    $pipelines = @{
+        "sdk_cli_tests" = 4;
+        "sdk_cli_global_config_tests" = 1;
+        "sdk_cli_azure_test" = 4;
+    }
+
+    $pipelines_success_count = @{
+        "sdk_cli_tests" = 0;
+        "sdk_cli_global_config_tests" = 0;
+        "sdk_cli_azure_test" = 0;
+    }
+    
+    $valid_status_array `
+    | ForEach-Object {
+        foreach ($key in $pipelines.Keys) {
+            $value = $pipelines[$key]
+            if ($value -eq 0) {
+                continue
+            }
+            if ($_.name.Contains($key)) {
+                if ($_.conclusion -ieq "success") {
+                    $pipelines_success_count[$key] += 1
+                } elseif ($_.conclusion -ieq "failure") {
+                    $failed_reason_ref.Value = "Required pipelines are not successful."
+                } else {
+                    if ($failed_reason_ref.Value -eq "") {
+                        $failed_reason_ref.Value = "Required pipelines are not finished."
+                    }
+                }
+            }
+            Write-Output "$($_.name) is $($_.conclusion)."
+        }
+    }
+}
+
+function run_checks() {
+
+    $need_to_check = New-Object System.Collections.Generic.HashSet[string]
+    git diff --name-only HEAD origin/main | ForEach-Object {
+        if ($snippet_debug -eq 1) {
+            Write-Output "git diff --name-only HEAD main $_"
+        }
+        if ($_.Contains("src/promptflow")) {
```

Comment:
seems broad now, shall we follow the code owner's for the check of SDK/CLI?

Reply:
This follows the trigger of SDK CLI :D

Reply:
by the way, we also monitor executor changes since their chane will affect the output of sdk...
CODEOWNER is not valid in this case.

Reply:
make sense, as SDK/CLI wrapped executor; maybe we can refine this once executor onboard this.